### ISSUE TITLE
Day of the week is behind 1 day

### ISSUE THREAD

**Issue by vatana7** [web-ridge/react-native-paper-dates#401]
### Current behaviour
Today is Fri 12 July 2024, however I am seeing Thur 12 July 2024 instead

### Expected behaviour
To see correct date 

### How to reproduce?
Just import `DatePickerInput` and use it normally

### Preview
<img width="336" alt="image" src="https://github.com/user-attachments/assets/9e0ffaf6-45fd-475e-aa07-733c3c916f93">

### What have you tried so far?
I have tried: 
- Updating the react-native-paper-dates package to latest version 

I cannot afford to update react native paper as it might bring unwanted behavior to my app 

### Your Environment

| software                      | version
| --------------------- | -------
| ios                                | 17.5
| android                        | x
| react-native                | 0.72.10
| react-native-paper    | 5.12.3
| node                            | 18.19.1
| npm or yarn                | yarn 1.22.21
| expo sdk                     | 49.0.23

**Comment by vatana7** [comment]
Here's more context:

Code:

```
import { useState } from 'react';
import { SafeAreaView, StyleSheet, View } from 'react-native';
import { DatePickerInput } from 'react-native-paper-dates';

export function CreateCustomEvent() {
  const [date, setDate] = useState(new Date());
  return (
    <SafeAreaView style={styles.root}>
      <View style={styles.container}>
        <DatePickerInput value={date} onChange={setDate} />
      </View>
    </SafeAreaView>
  );
}
const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  container: {
    padding: 24,
    flexGrow: 1,
  },
  contactRoot: {
    marginVertical: 24,
  },
  contactTitle: {
    fontWeight: '600',
    fontSize: 16,
  },
});

```

![Simulator Screen Recording - iPhone 15 Pro Max - 2024-07-12 at 20 03 20](https://github.com/user-attachments/assets/a259e708-b2af-4e88-a596-cef29f574259)

**Comment by vatana7** [comment]
Interesting, thank you for your reply. I'll debug a bit more I guess

**Comment by iM-GeeKy** [comment]
No problem, you should be able to use that snack with `new Date()` to see it work as expected. I tried it and it appeared to show the right date, so I'm under the impression perhaps it's an issue with the date you are passing. I felt like I've seen this one before in a previous closed issue.

**Comment by vatana7** [comment]
@iM-GeeKy so I found the problem, it has something to do with timezone. 

I ran this statement Intl.DateTimeFormat().resolvedOptions().timeZone and I expect it to be UTC+7 which is my timezone but it instead return UTC.

I guess my Hermes engine project, doesn't expose machine's timezone to it that's why I am seeing a day off. 

I have resolve the issue by using expo-localization's getCalendar() function to get proper timezone then setting the timezone with formatjs via this (documentation)[https://formatjs.io/docs/polyfills/intl-datetimeformat/#default-timezone] as well. 

I hope my comment help anyone encounter this issue.
